### PR TITLE
Add group name to ListDashboards response

### PR DESCRIPTION
**Pull request by sultan-duisenbay**
We need API to tell us if a dashboard belongs to a group. This functionality has been added to ListDashboards rpc.

### Notable changes
- Changed `ListDashboard` -> `ListDashboards`, `ListDashboardGroup` -> `ListDashboardGroups` and associated req / resp messages
- Added a new `DashboardResource` type
- `ListDashboards` now returns the dashboard group for any dashboard which belongs to one

### CODE REVIEW

**Review by chases2 — APPROVED** [review]
/hold

Overall LGTM; but we might want to move some of the heavier logic into loading the config for efficiency's sake (or include a TODO to make that improvement)

File: pkg/api/v1/config.go [hunk 1]
```diff
@@ -126,19 +126,27 @@ func (s Server) GetDashboardGroupHTTP(w http.ResponseWriter, r *http.Request) {
...
+func (s *Server) ListDashboards(ctx context.Context, req *apipb.ListDashboardsRequest) (*apipb.ListDashboardsResponse, error) {
 	c, err := s.getConfig(ctx, logrus.WithContext(ctx), req.GetScope())
 	if err != nil {
 		return nil, err
 	}
 	c.Mutex.RLock()
 	defer c.Mutex.RUnlock()
 
-	var resp apipb.ListDashboardResponse
+	dashboardsToGroups := make(map[string]string)
```
[the remark this answers is not in the document]
Comment (sultan-duisenbay) [reply pkg/api/v1/config.go:137]: Sounds good - I will add a TODO for that

[the remark this answers is not in the document]
Comment (sultan-duisenbay) [reply pkg/api/v1/config.go:138]: Sounds good - I will add a TODO for that

**Pull request merged**